core/lib/types/src/kzg.rs

```diff
@@ -0,0 +1,315 @@
+
+        let mut opening_value = [0u8; 32];
+        opening_value.copy_from_slice(&data[ptr..ptr + 32]);
+        ptr += 32;
+
+        let mut opening_proof = [0u8; 48];
+        opening_proof.copy_from_slice(&data[ptr..ptr + 48]);
+        ptr += 48;
+
+        let mut versioned_hash = [0u8; 32];
+        versioned_hash.copy_from_slice(&data[ptr..ptr + 32]);
+        ptr += 32;
+
+        let mut blob_proof = [0u8; 48];
+        blob_proof.copy_from_slice(&data[ptr..ptr + 48]);
+        ptr += 48;
+
+        assert_eq!(ptr, Self::SERIALIZED_SIZE);
+
+        Self {
+            blob: Blob::new(blob),
+            kzg_commitment: Bytes48::from_bytes(&kzg_commitment).unwrap(),
+            opening_point: Bytes32::new(opening_point),
+            opening_value: Bytes32::new(opening_value),
+            opening_proof: Bytes48::from_bytes(&opening_proof).unwrap(),
+            versioned_hash: Bytes32::new(versioned_hash),
+            blob_proof: Bytes48::from_bytes(&blob_proof).unwrap(),
+        }
+    }
+
+    /// Converts `KzgInfo` struct into a byte array
+    pub fn to_bytes(&self) -> [u8; Self::SERIALIZED_SIZE] {
+        let mut res = [0u8; Self::SERIALIZED_SIZE];
+
+        let mut ptr = 0;
+
+        res[ptr..ptr + BYTES_PER_BLOB].copy_from_slice(self.blob.as_slice());
+        ptr += BYTES_PER_BLOB;
+
+        res[ptr..ptr + 48].copy_from_slice(self.kzg_commitment.as_slice());
+        ptr += 48;
+
+        res[ptr..ptr + 32].copy_from_slice(self.opening_point.as_slice());
+        ptr += 32;
+
+        res[ptr..ptr + 32].copy_from_slice(self.opening_value.as_slice());
+        ptr += 32;
+
+        res[ptr..ptr + 48].copy_from_slice(self.opening_proof.as_slice());
+        ptr += 48;
+
+        res[ptr..ptr + 32].copy_from_slice(self.versioned_hash.as_slice());
+        ptr += 32;
+
+        res[ptr..ptr + 48].copy_from_slice(self.blob_proof.as_slice());
+        ptr += 48;
+
+        assert_eq!(ptr, Self::SERIALIZED_SIZE);
+
+        res
+    }
+
+    /// Construct all the KZG info we need for turning a piece of zksync pubdata into a 4844 blob.
+    /// The information we need is:
+    ///     1. zksync blob <- `pad_right`(pubdata)
+    ///     2. linear hash <- hash(zksync blob)
+    ///     3. 4844 blob <- `zksync_pubdata_into_ethereum_4844_data`(zksync blob)
+    ///     4. 4844 `kzg` commitment <- `blob_to_kzg_commitment`(4844 blob)
+    ///     5. versioned hash <- hash(4844 `kzg` commitment)
+    ///     6. opening point <- keccak(linear hash || versioned hash)[16..]
+    ///     7. opening value, opening proof <- `compute_kzg_proof`(4844)
+    ///     8. blob proof <- `compute_blob_kzg_proof`(blob, 4844 `kzg` commitment)
+    pub fn new(kzg_settings: &KzgSettings, pubdata: Vec<u8>) -> Self {
+        assert!(pubdata.len() <= BYTES_PER_BLOB_ZK_SYNC);
+
+        let mut zksync_blob = [0u8; BYTES_PER_BLOB_ZK_SYNC];
+        zksync_blob[0..pubdata.len()].copy_from_slice(&pubdata);
+
+        let mut hasher = Keccak256::new();
```

Comment:
since we have `sha256_hasher` in this function, let's call it `keccak256_hasher`